src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -137,22 +149,23 @@ public boolean canFullFillRecipe(final int qty, @NotNull final IItemHandler... i
         final int neededMultiplier = CraftingUtils.calculateMaxCraftingCount(qty, this);
         final List<ItemStorage> items = getCleanedInput();
 
-        for (final ItemStorage stack : items)
+        for (final ItemStorage storage : items)
         {
+            final ItemStack stack = storage.getItemStack();
```

Comment:
I get why you're doing this, and overall I love this change. I'm not looking forward to the merge conflict with my version that adds multiple outputs and predetermined secondary outputs based asking the IRecipe for them, etc.